test-cypress/integration/explore/visual.graph.spec.js

```diff
@@ -1,357 +1,203 @@
+        cy.get('.incontext-search-rdf-resource input').should('be.visible');
     });
 
-    it('Test expand collapsed node which has connections with double click', function () {
-        // Search for an valid resource
-        searchForResource(validResource);
-        // Verify redirection to existing visual graph
-        cy.url({timeout: 5000})
-            .should('include', validResource);
-
-        // Hover over the node with the mouse
-        cy.get('g[id="http://www.w3.org/TR/2003/PR-owl-guide-20031209/wine#' + validResource + '"] > circle', {
-            waitForAnimations: true,
-            animationDistanceThreshold: 0
-        })
-        // timeout is needed because mouseover event will result in
-        // pop-up of menu icons only if nodes are not moving
-            .should('be.visible', {timeout: 5000}).wait(5000)
-            .trigger('mouseover').wait(1000);
+    it('Test expand collapsed node which has connections with double click', () => {
+        searchForResource(VALID_RESOURCE);
 
-        // Select collapse function
-        cy.get('g.nodes-container > g.menu-events > g.collapse-icon > circle')
-            .trigger('click', {force: true});
+        getTargetNode().trigger('mouseover');
+        collapseGraph();
         // Verify that all links to the USRegion node are collapsed
-        cy.get('g.nodes-container > g.link-wrapper')
-            .should('have.length', '0');
+        getPredicates().should('have.length', 0);
         // Verify that the USRegion node is the only node left in the graph
-        cy.get('g.nodes-container > g.node-wrapper')
-            .should('have.length', '1').and('contain', 'USRegion');
+        getNodes().should('have.length', 1).and('contain', 'USRegion');
 
         // Double click on collapsed node
-        cy.get('g[id="http://www.w3.org/TR/2003/PR-owl-guide-20031209/wine#' + validResource + '"] > circle')
-            .click({force: true})
-            .then(() => {
-                cy.get('g[id="http://www.w3.org/TR/2003/PR-owl-guide-20031209/wine#' + validResource + '"] > circle')
-                    .click({force: true});
-            });
+        // This is ugly but unfortunately I couldn't make cypress's dblclick to work reliably here
+        getTargetNodeElement().click().then(() => {
+            getTargetNodeElement().click();
+        });
 
         // Verify that all links to the USRegion node are expanded
-        cy.get('g.nodes-container > g.link-wrapper')
-            .should('have.length', '2');
+        getPredicates().should('have.length', 2);
         // Verify that the USRegion node is not the only node left in the graph
-        cy.get('g.nodes-container > g.node-wrapper')
-            .its('length')
-            .should('be.gt', '1');
+        getNodes().should('have.length', 3);
     });
 
-    it('Test verify mouse/keyboard actions', function () {
+    it('Test verify mouse/keyboard actions', () => {
```

Comment:
I guess in the future we can make this check prettier.

Reply:
Yes, in fact this could be verified simply with `cy.get('.hotkeys-container').should('contain', 'Single click').and('contain', 'Rotate the graph to the right');`